src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -1927,16 +1932,17 @@ private static bool IsUserDefinedTrueOrFalse(BoundUnaryOperator @operator)
                 // is right. For example, C# allows the assignment statement:
                 //    (i) = 9;  
                 // So I don't assume this code should special case parenthesized expressions.
-                TypeSymbol type = null, convertedType = null;
+                TypeSymbol type = null, convertedType;
+                NullabilityInfo nullability = boundExpr.TopLevelNullability, convertedNullability;
                 Conversion conversion;
 
                 if (boundExpr.HasExpressionType())
                 {
                     type = boundExpr.Type;
 
-                    switch (boundExpr.Kind)
+                    switch (boundExpr)
                     {
-                        case BoundKind.Local:
+                        case BoundLocal local:
```

Comment:
>case BoundLocal local: [](start = 24, length = 22)

I am not sure what is the motivation for switching on a type.

Reply:
Like the other case, I would need to save the `BoundLocal` off to a local variable to get at the nullability, so I refactored this to have that local introduced for the scope of the relevant case.

---
In reply to: [269792020](https://github.com/dotnet/roslyn/pull/34429#discussion_r269792020) [](ancestors = 269792020)